### PR TITLE
Add bash script to run local e2e tests in parallel

### PR DESCRIPTION
**Pull request by jon-shipley**
* requires `parallel`

### CODE REVIEW

File: test/admin-hpa/run-local-tests.sh [hunk 1]
```diff
@@ -0,0 +1,83 @@
+#!/bin/sh
+#set -x
+
+# Run the tests locally in parallel
+JOB_COUNT=6
```
Comment (GuyHarwood) [inline test/admin-hpa/run-local-tests.sh:5]: would be nice to make this an optional param with a default of 6, or is there a particular reason you chose this?

File: test/admin-hpa/run-local-tests.sh [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+writeSuccess ()
+{
+    MSG=$1
+  echo -e "\033[92m ${MSG} \033[0m"
+}
+
+test ()
+{
+  FILE=$1
```
Comment (GuyHarwood) [inline test/admin-hpa/run-local-tests.sh:58]: can you document these params at the top of the file please, and also check they exist?

File: test/pupil-hpa/run-local-tests.sh [hunk 3]
```diff
@@ -0,0 +1,78 @@
+#!/bin/sh
+
```
Comment (GuyHarwood) [inline test/pupil-hpa/run-local-tests.sh:2]: same comments apply to this as the admin one...

**Review by GuyHarwood — CHANGES_REQUESTED** [review]
superb bit of work 🏆  this will make a huge difference

just a few comments that would improve the script for someone who's not used it before

File: test/admin-hpa/run-local-tests.sh (hunk 1 above)
Comment (jon-shipley) [reply test/admin-hpa/run-local-tests.sh:5]: 6 seems a good choice for an 8 core system, and was chosen after a lot of tests runs from 5-10.  It works well.  If anyone has wildly more cores we can make it a param.

File: test/admin-hpa/run-local-tests.sh (hunk 2 above)
Comment (jon-shipley) [reply test/admin-hpa/run-local-tests.sh:58]: 😀. This is the documentation.

In a bash function $1 refers to the 1st parameter passed to the function, $2 to the 2nd and so on.   So while it's possible to just use $1 or $2 within the function body, this assignment at least makes it readable.

Comment (GuyHarwood) [reply test/admin-hpa/run-local-tests.sh:58]: yes, i am aware of the parameter indexing and $0 being the script itself.  

If you check for the existence of any required params near the top of the file and inform the user if anything is missing, it explains itself and stops it running without anything missing.

Here is [an example](https://github.com/DFEAGILEDEVOPS/MTC/blob/master/deploy/keys/function-masterkey-update-keyvault.sh)

Comment (GuyHarwood) [reply test/admin-hpa/run-local-tests.sh:58]: It's no dealbreaker, so if you're averse to such an addition just leave as is...

Comment (GuyHarwood) [reply test/admin-hpa/run-local-tests.sh:58]: i've now realised after discussion that these are local function arguments, not script arguments.
resolving this...

**Review by GuyHarwood — APPROVED** [review]
LGTM

**Pull request merged**